crates/dbsp/src/operator/distinct.rs

```diff
@@ -23,6 +23,53 @@ use std::{
 circuit_cache_key!(DistinctId<C, D>(GlobalNodeId => Stream<C, D>));
 circuit_cache_key!(DistinctIncrementalId<C, D>(GlobalNodeId => Stream<C, D>));
 
+impl<C, D> Stream<C, D>
+where
+    C: Circuit,
+    D: 'static,
+{
+    /// Marks the data within the current stream as distinct, meaning that all
+    /// further calls to `.distinct()` will have no effect.
+    ///
+    /// This must only be used on streams whose integral contain elements with
+    /// unit weights only, otherwise this will cause the dataflow to yield
+    /// incorrect results
+    pub fn mark_distinct(&self) -> Self {
```

Comment:
This is something that is best done by the compiler

Reply:
This shouldn't get in the way of compiler optimizations when we have them. If it does, we can adjust the API.